generator/src/main/java/com/scottlogic/deg/generator/decisiontree/DecisionTreeSimplifier.java

```diff
@@ -68,22 +68,19 @@ private ConstraintNode simplifySingleOptionDecisions(ConstraintNode node) {
                 (parentConstraint, decisionNode) -> {
                     ConstraintNode firstOption = decisionNode.getOptions().iterator().next();
                     if (parentConstraint.getAtomicConstraints().stream().anyMatch(firstOption.getAtomicConstraints()::contains)) {
```

Comment:
it might be worth seeing if this looks better like

```
builder = parentConstraint.builder.removedecision.

if ( not (the current if) {
    builder = builder.addAtomicConstraints(..).addDecisions(...)

return builder.build.

```

Reply:
As discussed I am not sure I like the idea of not immediately terminating builders.